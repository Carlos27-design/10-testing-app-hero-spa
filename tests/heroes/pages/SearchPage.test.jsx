import { MemoryRouter } from 'react-router-dom';
import { SearchPage } from '../../../src/heroes';
import { fireEvent, render, screen } from '@testing-library/react';

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUseNavigate,
}));

describe('Pruebas en <SearchPage/>', () => {
  test('debe de mostrarse con valores por defecto', () => {
    const { container } = render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar a batman y el input con el valor del queryString', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=batman']}>
        <SearchPage />
      </MemoryRouter>
    );
    const input = screen.getByRole('textbox');
    expect(input.value).toBe('batman');

    const img = screen.getByRole('img');
    expect(img.src).toContain('/assets/heroes/dc-batman.jpg');

    const alert = screen.getByLabelText('alert-danger');

    expect(alert.style.display).toBe('none');
  });

  test('debe de mostrar un error si no se encuentra el heroe', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=batman123123']}>
        <SearchPage />
      </MemoryRouter>
    );

    const alert = screen.getByLabelText('alert-danger');

    expect(alert.style.display).toBe('');
  });

  test('debe de llamar al el navigate a la pantalla nueva', () => {
    render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    );

    const buscar = screen.getByRole('textbox');
    fireEvent.change(buscar, { target: { value: 'Superman' } });

    const buscarBtn = screen.getByLabelText('search');
    fireEvent.click(buscarBtn);

    expect(mockedUseNavigate).toHaveBeenCalledWith('?q=Superman');
  });
});
