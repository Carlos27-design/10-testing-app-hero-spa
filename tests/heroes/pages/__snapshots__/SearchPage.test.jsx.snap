// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage/> debe de mostrarse con valores por defecto 1`] = `
<div>
  <div
    class="row"
  >
    <h1>
      SearchPage
    </h1>
    <hr />
    <div
      class="col-5"
    >
      <h4>
        Rearching
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          aria-label="search"
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search a hero!!!!!
      </div>
      <div
        aria-label="alert-danger"
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        No hero with 
        <b />
      </div>
    </div>
  </div>
</div>
`;
